apps/web/src/utils/formInstanceUtils.ts

```diff
@@ -83,3 +88,23 @@ export const nextSigner = (formInstance: FormInstanceEntity) => {
 
   return firstUnsignedSignature;
 };
+
+/**
+ * Determines if a signature's next signer is the current user
+ *
+ * @param signature the signature to check
+ * @param user the current user
+ * @returns true if the next signer is the current user, false otherwise
+ */
+export const signerIsUser = (signature: SignatureEntity, user: User) => {
+  if (!signature) return false;
+
+  const signerType = signature.signerType as any;
+  return (
+    (signerType === 'USER' && signature.assignedUserId === user?.id) ||
+    (signerType === 'POSITION' &&
+      signature.signerPositionId === user?.positionId) ||
+    (signerType === 'DEPARTMENT' &&
+      user?.departmentId === signature.signerDepartmentId)
```

Comment:
Maybe check if user is not null here as well so you don't have to do the "user?"